src/Renci.SshNet.Tests/Classes/ShellStreamTest.cs

```diff
@@ -103,6 +103,36 @@ public void WriteLine_Line_ShouldOnlyWriteLineTerminatorWhenLineIsNull()
             _channelSessionMock.Verify(p => p.SendData(lineTerminator), Times.Once);
         }
 
+        [TestMethod]
+        public void WindowChangeRequest_ShouldReturnFalseWhenNotConnected()
+        {
+            var shellStream = CreateShellStream();
+            _channelSessionMock.Setup(s => s.IsOpen).Returns(false);
+            Assert.IsFalse(shellStream.WindowChangeRequest(80, 25, 0, 0));
+        }
+
+        [TestMethod]
+        public void WindowChangeRequest_ShouldReturnFalseWhenResultFails()
+        {
+            var shellStream = CreateShellStream();
+            _channelSessionMock.Setup(s => s.IsOpen).Returns(true);
+            _channelSessionMock.Setup(s => s.SendWindowChangeRequest(
+                  It.IsAny<uint>(), It.IsAny<uint>(),
+                  It.IsAny<uint>(), It.IsAny<uint>())).Returns(false);
+            Assert.IsFalse(shellStream.WindowChangeRequest(80, 25, 0, 0));
+        }
+
+        [TestMethod]
+        public void WindowChangeRequest_ShouldReturnTrueWhenResultSucceeds()
+        {
+            var shellStream = CreateShellStream();
+            _channelSessionMock.Setup(s => s.IsOpen).Returns(true);
+            _channelSessionMock.Setup(s => s.SendWindowChangeRequest(
+                  It.IsAny<uint>(), It.IsAny<uint>(),
+                  It.IsAny<uint>(), It.IsAny<uint>())).Returns(true);
+            Assert.IsTrue(shellStream.WindowChangeRequest(80, 25, 0, 0));
```

Comment:
Please also verify that `SendWindowChangeRequest(...)` was actually invoked (once).